File: app/src/main/java/at/droelf/travellogapp/DateTimeUtils.java
package at.droelf.travellogapp;

import org.joda.time.DateTime;
import org.joda.time.LocalDateTime;
import org.joda.time.format.ISODateTimeFormat;

public class DateTimeUtils {

    static String dateTimeToIsoString(DateTime dateTime) {
        return ISODateTimeFormat.dateTimeNoMillis().print(dateTime);
    }

    static DateTime iosStringToDateTime(String isoString) {
        return ISODateTimeFormat.dateTimeNoMillis().parseDateTime(isoString);
    }

    static String localDateTimeToIsoString(LocalDateTime dateTime) {
        return ISODateTimeFormat.dateTimeNoMillis().print(dateTime);
    }

    static LocalDateTime iosStringToLocalDateTime(String isoString) {
        return ISODateTimeFormat.dateTimeNoMillis().parseLocalDateTime(isoString+"Z");
    }

}

File: app/src/main/scala/at/droelf/travellogapp/DateTimeUtils.scala
package at.droelf.travellogapp

import org.joda.time.DateTime;
import org.joda.time.LocalDateTime;
import org.joda.time.format.{DateTimeFormat, DateTimeFormatter, ISODateTimeFormat}
;

object DateTimeUtils {

    def dateTimeToIsoString(dateTime: DateTime) = ISODateTimeFormat.dateTimeNoMillis().print(dateTime)

    def isoStringToDateTime(isoString: String) = ISODateTimeFormat.dateTimeNoMillis().parseDateTime(isoString)

    def localDateTimeToIsoString(dateTime: LocalDateTime) = ISODateTimeFormat.dateTimeNoMillis().print(dateTime)

    def isoStringToLocalDateTime(isoString: String) = ISODateTimeFormat.dateTimeNoMillis().parseLocalDateTime(isoString+"Z")

    def parseExifDateToLocalDateTime(exifDate: String): LocalDateTime = {
        val format = DateTimeFormat.forPattern("yyyy:MM:dd HH:mm:ss")
        LocalDateTime.parse(exifDate, format)
    }

}
